lib/widgets/pin_code_widget.dart

```diff
@@ -26,27 +26,53 @@ class PinCodeWidget extends StatefulWidget {
   }
 }
 
-class PinCodeWidgetState extends State<PinCodeWidget> {
+class PinCodeWidgetState extends State<PinCodeWidget>
+    with WidgetsBindingObserver {
   String _enteredPinCode;
   String _errorMessage;
+  String _enrolledBiometrics;
 
   @override
   initState() {
     super.initState();
+    WidgetsBinding.instance.addObserver(this);
     _enteredPinCode = "";
     _errorMessage = "";
+    _enrolledBiometrics = "";
     if (widget.onFingerprintEntered != null) {
-      widget.userProfileBloc.userStream.first.then(
-        (user) async {
-          if (user.securityModel.enrolledBiometrics != "") {
-            await Future.delayed(Duration(milliseconds: 240));
-            if (this.mounted) _validateBiometrics();
-          }
-        },
-      );
+      _promptBiometrics();
     }
   }
 
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    if (state == AppLifecycleState.resumed) {
+      if (widget.onFingerprintEntered != null) {
```

Comment:
`onFingerprintEntered` is true when the settings are true but the check for if the biometrics are enables is missing here, isn't it?